### PR TITLE
boards: add Ublox C030-U201

### PR DESCRIPTION
**Pull request by vincent-d**
### Contribution description

This adds the support of the [Ublox C030-U201](https://www.u-blox.com/en/product/c030-app-board) board. This is a stm32f4-based board (the cpu support is also added here) with a Ublox 3G modem (SARA-U201) and GPS (MAX-M8C).
Most of the peripheral are added from the datasheet/schematics. I tested the stdio UART, LEDs and button through saul and xtimer.

### Issues/PRs references

Can be a good base to work on #8602

### CODE REVIEW

File: boards/c030-u201/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,10 @@
+USEMODULE += stm32_periph_uart_hw_fc
+
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += saul_gpio
+  # USEMODULE += si7034 # TODO: add when si70xx driver is updated
```
Comment (aabadie) [inline boards/c030-u201/Makefile.dep]: What is missing in this driver ?

File: cpu/stm32f4/include/cpu_conf.h [hunk 2]
```diff
@@ -49,6 +51,10 @@
 #include "vendor/stm32f446xx.h"
 #elif defined(CPU_MODEL_STM32F429ZI)
 #include "vendor/stm32f429xx.h"
+#elif defined(CPU_MODEL_STM32F437VG)
+#include "vendor/stm32f437xx.h"
+#elif defined(CPU_MODEL_STM32F446RE) || defined(CPU_MODEL_STM32F446ZE)
```
Comment (aabadie) [inline cpu/stm32f4/include/cpu_conf.h]: Duplicate of line 50

File: cpu/stm32f4/include/periph_cpu.h [hunk 3]
```diff
@@ -49,7 +49,7 @@ enum {
  * @brief   Available number of ADC devices
  */
 #if defined(CPU_MODEL_STM32F401RE) || defined(CPU_MODEL_STM32F410RB) \
-    || defined(CPU_MODEL_STM32F411RE) \
+    || defined(CPU_MODEL_STM32F411RE) || defined(CPU_MODEL_STM32F412ZG) \
```
Comment (aabadie) [inline cpu/stm32f4/include/periph_cpu.h]: Looks unrelated, is this a pending bug ?

File: cpu/stm32f4/include/periph_cpu.h [hunk 4]
```diff
@@ -59,7 +59,9 @@ enum {
     || defined(CPU_MODEL_STM32F423MH) || defined(CPU_MODEL_STM32F423VH) \
     || defined(CPU_MODEL_STM32F423ZH)
 #define ADC_DEVS            (1U)
-#elif defined(CPU_MODEL_STM32F407VG) || defined(CPU_MODEL_STM32F415RG) || defined(CPU_MODEL_STM32F446RE) || defined(CPU_MODEL_STM32F429ZI)
+#elif defined(CPU_MODEL_STM32F407VG) || defined(CPU_MODEL_STM32F415RG) \
+    || defined(CPU_MODEL_STM32F429ZI) || defined(CPU_MODEL_STM32F437VG) \
+    || defined(CPU_MODEL_STM32F446RE) || defined(CPU_MODEL_STM32F446ZE)
```
Comment (aabadie) [inline cpu/stm32f4/include/periph_cpu.h]: Also looks unrelated.

File: cpu/stm32f4/include/periph_cpu.h (hunk 3 above)
Comment (vincent-d) [reply cpu/stm32f4/include/periph_cpu.h]: Not sure, I'm think I messed up when rebasing. I'll take a look.

File: boards/c030-u201/Makefile.dep (hunk 1 above)
Comment (vincent-d) [reply boards/c030-u201/Makefile.dep]: I'm not sure to remember, but there are 2 sub-families and `si7034` is not in the one which was imlpemented.

**Comment by vincent-d** [comment]
- Removed unrelated changes and opened #9175
- renamed board to `ublox-c030-u201`

**Comment by aabadie** [comment]
#9175 is merged and now you need to rebase...

**Comment by vincent-d** [comment]
Rebased

**Comment by vincent-d** [comment]
> There's a typo in the board directory name

Oops, fixed

File: cpu/stm32f4/include/periph_cpu.h [hunk 5]
```diff
@@ -60,8 +60,8 @@ enum {
     || defined(CPU_MODEL_STM32F423ZH)
 #define ADC_DEVS            (1U)
 #elif defined(CPU_MODEL_STM32F407VG) || defined(CPU_MODEL_STM32F415RG) \
-    || defined(CPU_MODEL_STM32F429ZI) || defined(CPU_MODEL_STM32F446RE) \
-    || defined(CPU_MODEL_STM32F446ZE)
+    || defined(CPU_MODEL_STM32F429ZI) || defined(CPU_MODEL_STM32F437VG) \
+    || defined(CPU_MODEL_STM32F446RE)
```
Comment (aabadie) [inline cpu/stm32f4/include/periph_cpu.h]: Now the f446ze is missing here :)

File: cpu/stm32f4/include/periph_cpu.h [hunk 6]
```diff
@@ -49,7 +49,7 @@ enum {
  * @brief   Available number of ADC devices
  */
 #if defined(CPU_MODEL_STM32F401RE) || defined(CPU_MODEL_STM32F410RB) \
-    || defined(CPU_MODEL_STM32F411RE) || defined(CPU_MODEL_STM32F412ZG) \
+    || defined(CPU_MODEL_STM32F411RE) \
```
Comment (aabadie) [inline cpu/stm32f4/include/periph_cpu.h]: F412 missing

**Comment by vincent-d** [comment]
Should be OK now, sorry.

**Comment by vincent-d** [comment]
Squashed

**Comment by aabadie** [comment]
Murdock raises other problems, probably raised by uninitialized attributes in periph conf struct. You may have to set them for some boards.

**Comment by vincent-d** [comment]
This is a very annoying bug from g++ with c++11...it should be fixed. I amended directly.

**Comment by dylad** [comment]
This PR violates the ongoing embargo on I2C.
